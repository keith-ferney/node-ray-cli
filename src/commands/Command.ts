/* eslint-disable no-unused-vars */

import { classOf } from '@/lib/helpers';
import { Argv } from 'yargs';
import { Ray } from 'node-ray';

export abstract class Command {
    public command = '';

    public help = '';

    public static instance: Command;

    public client: any = null;

    public uuid: string | null = null;

    public argv: Argv | null = null;

    constructor() {
        Command.instance = this;
    }

    public name(): string {
        return <string>this.command.split(' ').shift();
    }

    public builder = (): Record<string, any> => {
        return {};
    };

    public handle(argv: Argv) {
        this.argv = argv;
    }

    public displayUuid(instance: Ray) {
        if (this.argv === null) {
            return;
        }

        if (typeof this.argv['quiet'] === 'undefined' || this.argv['quiet'] === false) {
            console.log(instance.uuid);
        }
    }
}
