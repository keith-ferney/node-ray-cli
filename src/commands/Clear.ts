import { Command } from './Command';
import { Argv } from 'yargs';
import { Ray } from 'node-ray';

export class Clear extends Command {
    public override command = 'clear';
    public override help = 'clear the current Ray screen';

    public override handle(argv: Argv) {
        super.handle(argv);

        const instance = Ray.create(this.client, this.uuid);

        instance.clearScreen();
    }
}
