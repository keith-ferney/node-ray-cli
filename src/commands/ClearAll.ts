import { Command } from './Command';
import { Argv } from 'yargs';
import { Ray } from 'node-ray';

export class ClearAll extends Command {
    public override command = 'clear-all';
    public override help = 'Clear current and all previous Ray screens';

    public override handle(argv: Argv) {
        if (super.handle(argv) === false) {
            return false;
        }

        const instance = Ray.create(this.client, this.uuid ?? argv['uuid'] ?? null);

        instance.clearAll();

        super.afterHandle(instance);
    }
}
