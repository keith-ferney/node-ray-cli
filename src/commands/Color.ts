import { Command } from './Command';
import { Argv } from 'yargs';
import { Ray } from 'node-ray';

export class Color extends Command {
    public override command = 'color <uuid> <color>';
    public override help = 'change to color of a previously sent payload';

    public override handle(argv: Argv) {
        super.handle(argv);

        const instance = Ray.create(this.client, argv['uuid']);

        instance.uuid = argv['uuid'];

        instance.color(argv['color']);

        this.displayUuid(instance);
    }
}
