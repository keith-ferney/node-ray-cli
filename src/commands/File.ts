import { Command } from './Command';
import { Argv } from 'yargs';
import { Ray } from 'node-ray';

export class File extends Command {
    public override command = 'file <filename>';
    public override help = 'Display the contents of a file in Ray';

    public override handle(argv: Argv) {
        if (super.handle(argv) === false) {
            return false;
        }

        const instance = Ray.create(this.client, this.uuid ?? argv['uuid']);

        instance.file(argv['filename']);

        super.displayUuid(instance);

        super.afterHandle(instance);
    }
}
