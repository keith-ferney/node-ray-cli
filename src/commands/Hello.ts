import { Command } from './Command';
import { Argv } from 'yargs';

export class Hello extends Command {
    public override command = 'hello <name>';
    public override help = 'say hello to someone';

    public override handle(argv: Argv) {
        console.log(`hello ${argv.name}!`);
    }
}
