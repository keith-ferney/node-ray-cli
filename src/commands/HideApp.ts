import { Command } from './Command';
import { Argv } from 'yargs';
import { Ray } from 'node-ray';

export class HideApp extends Command {
    public override command = 'hide-app';
    public override help = 'Hide the Ray application';

    public override handle(argv: Argv) {
        if (super.handle(argv) === false) {
            return false;
        }

        const instance = Ray.create(this.client, this.uuid);

        instance.hideApp();
    }
}
