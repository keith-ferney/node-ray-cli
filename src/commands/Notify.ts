import { Command } from './Command';
import { Argv } from 'yargs';
import { Ray } from 'node-ray';

export class Notify extends Command {
    public override command = 'notify <message>';
    public override help = 'Display a desktop notification from Ray';

    public override handle(argv: Argv) {
        if (super.handle(argv) === false) {
            return false;
        }

        const instance = Ray.create(this.client, this.uuid ?? argv['uuid']);

        instance.notify(argv['message']);

        super.displayUuid(instance);
    }
}
