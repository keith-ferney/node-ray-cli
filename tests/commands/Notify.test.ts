/* eslint-disable no-undef */
import { Notify } from '@/commands/Notify';
import { FakeClient } from '@tests/TestClasses/FakeClient';
import { Argv } from 'yargs';

let client: FakeClient;
let command: Notify;

beforeEach(() => {
    client = new FakeClient();

    command = new Notify();
    command.uuid = 'fakeUuid';
    command.client = client;
});

it('has the correct name', () => {
    expect(command.name()).toBe('notify');
});

it('sends an image payload', () => {
    command.handle(<Argv>(<unknown>{ _: ['notify'], quiet: true, message: 'test message' }));

    expect(client.sentPayloads()).toMatchSnapshot();
});
