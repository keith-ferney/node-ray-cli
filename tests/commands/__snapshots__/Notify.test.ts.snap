// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sends an image payload 1`] = `
Array [
  Object {
    "meta": Array [],
    "payloads": Array [
      Object {
        "content": Object {
          "value": "test message",
        },
        "origin": Object {
          "file": "/src/commands/Notify.ts",
          "function_name": "xxxx",
          "hostname": "fake-host",
          "line_number": 999,
        },
        "type": "notify",
      },
    ],
    "uuid": "fakeUuid",
  },
]
`;
